sub-packages/bionemo-geneformer/tests/bionemo/geneformer/scripts/test_pydantic_train.py

```diff
@@ -85,16 +93,79 @@ def test_pretrain_cli_from_ckpt(tmpdir, data_path):
     assert (result_dir / "test-experiment").exists(), "Could not find test experiment directory."
 
 
+@pytest.mark.slow
+def test_pretrain_api(tmpdir, data_path):
+    from bionemo.geneformer.run.main import main
+
+    result_dir = Path(tmpdir.mkdir("results"))
+    config = f"{result_dir}/test_config.yaml"
+
+    args = argparse.Namespace(
+        recipe="geneformer_tiny_test_recipe",
+        dest=config,
+        data_path=str(data_path),
+        result_dir=result_dir,
+        initial_ckpt_path=None,
+    )
+    config = save_recipe(args)
+
+    args = argparse.Namespace(
+        config=config,
+        model_config_cls=ExposedGeneformerPretrainConfig,
+        data_config_cls=GeneformerPretrainingDataConfig,
```

Comment:
Same comment here making sure you cover testing how a user would enter a different setting. Do we want to support `Type` as input? The parser just says `Optional[str]`. Seems like this is a special case that works because it doesn't get re-parsed in the if/else block since a Type doesn't match the if/else options.